contracts/resolvers/DelegatableResolverFactory.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "./DelegatableResolver.sol";
+import {ClonesWithImmutableArgs} from "clones-with-immutable-args/src/ClonesWithImmutableArgs.sol";
+
+/**
+ * A resolver factory that creates a dedicated resolver for each user
+ */
+
+contract DelegatableResolverFactory {
+    using ClonesWithImmutableArgs for address;
+
+    DelegatableResolver public implementation;
+    event newDelegatableResolver(address indexed resolver, address owner);
+
+    constructor(DelegatableResolver implementation_) {
+        implementation = implementation_;
+    }
+
+    /*
+     * Create the unique address unique to the owner
+     * @param address The address of the resolver owner
+     * @return address The address of the newly created Resolver
+     */
+    function createClone2(
```

Comment:
The caller doesn't need to know the internal details of how resolvers are created.

```suggestion
    function create(
```